### PR TITLE
fix(route53-targets): deprecated method for dns name is used in userpool domain target (under feature flag)

### PR DESCRIPTION
**Pull request by go-to-k**
### Reason for this change

The [PR](https://github.com/aws/aws-cdk/pull/31402) created a new method to get CloudFront DNS name in Cognito user pool domain. A custom resource is created in the old method, but is not in the new method.

However, the `UserPoolDomainTarget` in the `route53-targets` module continues to use the old method. So we should change to use the new method.

### Description of changes

The `bind` method in the `UserPoolDomainTarget` implements `IAliasRecordTarget` interface, so a new method instead of the `bind` cannot be created.

Therefore, I take it using a feature flag.

### Description of how you validated changes

Both unit and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nmussy** [comment]
I didn't have a long enough look to give a meaningful review, but I'm not sure "new method" is descriptive enough. Not to mention that we might add "newer" method flags in the future

**Comment by go-to-k** [comment]
@nmussy 

You're absolutely right (I hadn't really thought about it properly...). I have corrected it as such.

File: packages/@aws-cdk-testing/framework-integ/test/aws-route53-targets/test/integ.userpool-domain-alias-target.ts
Comment (nmussy) [inline packages/@aws-cdk-testing/framework-integ/test/aws-route53-targets/test/integ.userpool-domain-alias-target.ts:34]: I was going to ask to add a test to make sure the CR wasn't created, but you're already checking for it in the unit test, so it might be overkill

https://github.com/aws/aws-cdk/pull/31403/files#diff-d39a13fcbdb9a308f6017d9800b5f4244d21103d76b708c68d84226d0172faeaR78-R79

**Review by nmussy — APPROVED** [review]
LGTM. It might be better to rename the feature flag to say what it does vs what it doesn't do anymore, ie. attribute referencing vs CR lookup, but I'll let maintainers weigh in on this

**Review by GavinZZ — APPROVED** [review]

**Pull request merged**
